druid/src/widget/button.rs

```diff
@@ -40,12 +37,52 @@ impl<T: Data> Button<T> {
     pub fn new(
         text: impl Into<LabelText<T>>,
         action: impl Fn(&mut EventCtx, &mut T, &Env) + 'static,
-    ) -> Button<T> {
-        Button {
-            label: Label::new(text),
-            label_size: Size::ZERO,
-            action: Box::new(action),
-        }
+    ) -> impl Widget<T> {
+        let painter = Self::painter();
+        CenteredLabel::new(text)
+            .padding(LABEL_INSETS)
+            .background(painter)
+            .on_click(action)
+    }
+
+    fn painter() -> Painter<T> {
+        Painter::new(|ctx, _, env| {
+            let is_active = ctx.is_active();
+            let is_hot = ctx.is_hot();
+            let size = ctx.size();
+
+            let rounded_rect = Rect::from_origin_size(Point::ORIGIN, size)
+                .inset(-1.0)
```

Comment:
The inset needs to be dynamic, because the border width is dynamic.
```rust
let border_width = env.get(theme::BUTTON_BORDER_WIDTH);
// ...
.inset(border_width / -2.0)
// ...
ctx.stroke([.......], border_width);
```

Reply:
Thank you!